### PR TITLE
Update prepare unit tests to use REST endpoints and cleanup

### PR DESCRIPTION
**Pull request by clark800**

### CODE REVIEW

File: api/lib/tx-to-rest-converter.js [hunk 1]
```diff
@@ -60,113 +60,109 @@ function parseFlagsFromResponse(responseFlags, flags) {
...
+  var order_changes = renameCounterpartyToIssuerInOrderChanges(
+    parseOrderBookChanges(meta))[account];
+
+  var partialPayment = (isPartialPayment(tx) && meta.DeliveredAmount) ? {
+      destination_amount_submitted: convertAmount(tx.Amount),
+      source_amount_submitted: convertAmount(tx.SendMax || tx.Amount)
+    } : {};
+
+  return _.assign({
+    result: tx.meta ? tx.meta.TransactionResult : '',
```
Comment (geertweening) [inline api/lib/tx-to-rest-converter.js]: Why `tx.meta` here versus `meta` in other places?

File: api/lib/validate.js [hunk 2]
```diff
@@ -529,10 +537,10 @@ function validateOptions(options) {
     validateLedger(options.ledger);
   }
   if (options.validated !== undefined && !isBoolean(options.validated)) {
-    throw error('validated must be boolean, not: ' + options.validated);
+    throw error('"validated" must be boolean, not: ' + options.validated);
```
Comment (geertweening) [inline api/lib/validate.js:536]: `typeof options.validated` ?

File: api/lib/validate.js [hunk 3]
```diff
@@ -529,10 +537,10 @@ function validateOptions(options) {
...
   }
   if (options.submit !== undefined && !isBoolean(options.submit)) {
-    throw error('submit must be boolean, not: ' + options.submit);
+    throw error('"submit" must be boolean, not: ' + options.submit);
```
Comment (geertweening) [inline api/lib/validate.js:539]: same here

File: api/payments.js [hunk 4]
```diff
@@ -113,89 +80,53 @@ function formatPaymentHelper(account, transaction, callback) {
...
 
-    if (Number(options.last_ledger_sequence) > 0) {
-      ledgerIndex = Number(options.last_ledger_sequence);
-    } else {
-      ledgerIndex = Number(remote._ledger_current_index)
-        + transactions.DEFAULT_LEDGER_BUFFER;
-    }
-
-    transaction.lastLedger(ledgerIndex);
+    _transaction.lastLedger(Number(options.last_ledger_sequence) ||
```
Comment (geertweening) [inline api/payments.js]: Since `0`, `NaN` and negative are falsy, this works. Consider casting to Boolean for clarity

File: api/payments.js [hunk 5]
```diff
@@ -113,89 +80,53 @@ function formatPaymentHelper(account, transaction, callback) {
...
-        transact(transaction, self, secret, _options, converter, _callback);
-      } catch (err) {
-        _callback(err);
-      }
-    }
-  ], callback);
+  var initialTx = createPaymentTransaction(account, payment);
+  var transaction = options.submit !== false ? prepareTransaction(
+    initialTx, this.remote) : initialTx;
+  var converter = options.submit !== false ? formatTransactionResponse :
```
Comment (geertweening) [inline api/payments.js]: Duplicating `options.submit !== false` 4 times here. Consider storing the result in a variable

File: server/response-handler.js [hunk 6]
```diff
@@ -75,16 +75,7 @@ function send(response, body, statusCode) {
  *                success property
```
Comment (geertweening) [inline server/response-handler.js:75]: Nice cleanup of the response handler

File: api/lib/tx-to-rest-converter.js (hunk 1 above)
Comment (clark800) [inline api/lib/tx-to-rest-converter.js]: Nice catch, I missed this one.

File: api/lib/validate.js (hunk 2 above)
Comment (clark800) [inline api/lib/validate.js:536]: Changed to `!_.isBoolean`, which is shorter than `typeof _ !== 'boolean'`

File: api/payments.js (hunk 4 above)
Comment (clark800) [inline api/payments.js]: Will move `||` inside `Number` call

File: api/payments.js (hunk 5 above)
Comment (clark800) [inline api/payments.js]: I was considering that too, will do

**Comment by geertweening** [comment]
LGTM

**Pull request merged**